File: OLSource1/v1/VS_officedev/codesnippet/CSharp/walkthrough--adding-controls-to-a-worksheet-at-run-time-in-vsto-add-in-project_3.cs
        private void ListObject_Click(object sender, RibbonControlEventArgs e)
        {
            Worksheet worksheet = Globals.Factory.GetVstoObject(
                Globals.ThisAddIn.Application.ActiveWorkbook.Worksheets[1]);


            string listObjectName = "MyListObject";

            if (((RibbonCheckBox)sender).Checked)
            {
                Excel.Range selection = Globals.ThisAddIn.Application.Selection as Excel.Range;
                if (selection != null)
                {
                    worksheet.Controls.AddListObject(selection, listObjectName);
                }
            }
            else
            {
                worksheet.Controls.Remove(listObjectName);
            }
        }

File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/walkthrough--adding-controls-to-a-worksheet-at-run-time-in-vsto-add-in-project_3.vb
    Private Sub ListObject_Click(ByVal sender As System.Object, _
        ByVal e As Microsoft.Office.Tools.Ribbon.RibbonControlEventArgs) _
            Handles ListObject.Click

        Dim NativeWorksheet As Microsoft.Office.Interop.Excel.Worksheet =
            Globals.ThisAddIn.Application.ActiveWorkbook.Worksheets(1)

        Dim worksheet As Microsoft.Office.Tools.Excel.Worksheet =
            Globals.Factory.GetVstoObject(NativeWorksheet)


        Dim listObjectName As String = "MyListObject"

        If CType(sender, RibbonCheckBox).Checked Then
            Dim selection As Excel.Range = _
                Globals.ThisAddIn.Application.Selection
            If Not (selection Is Nothing) Then
                worksheet.Controls.AddListObject(selection, listObjectName)
            End If
        Else
            worksheet.Controls.Remove(listObjectName)
        End If
    End Sub